runtime/lua/vim/lsp/buf.lua

```diff
@@ -243,11 +243,15 @@ local function get_locations(method, opts)
 
       vim.bo[b].buflisted = true
       local w = win
-      if opts.reuse_win then
+      if opts.reuse_win and vim.api.nvim_win_get_buf(w) ~= b then
         w = vim.fn.win_findbuf(b)[1] or w
-        if w ~= win then
-          api.nvim_set_current_win(w)
+        for _, tw in ipairs(vim.api.nvim_tabpage_list_wins(0)) do
+          if vim.api.nvim_win_get_buf(tw) == b then
```

Comment:
Can `vim.fn.bufwinid()` or some function like that be used to avoid some of this code?